tests/unittests/GraphOptzTest.cpp

```diff
@@ -28,11 +28,34 @@ using namespace glow;
 
 class GraphOptz : public ::testing::Test {
 public:
-  GraphOptz() { F_ = mod_.createFunction("main"); }
+  GraphOptz() {
+    mod_ = &EE_.getModule();
+    originalF_ = mod_->createFunction("main");
+  }
 
 protected:
-  Module mod_;
-  Function *F_;
+  void checkNumericalEquivalence() {
+    // Check that the function and its optimized complement exist.
+    EXPECT_TRUE(originalF_);
+    EXPECT_TRUE(optimizedF_);
+
+    // Clone bindings to use for original and optimized functions.
+    PlaceholderBindings originalBindings = bindings_.clone();
+    PlaceholderBindings optimizedBindings = bindings_.clone();
+
+    // Compile and run functions.
+    EE_.compile(CompilationMode::Infer);
+    EE_.run(originalBindings, originalF_->getName());
+    EE_.run(optimizedBindings, optimizedF_->getName());
+
+    // Compare outputs.
+    EXPECT_TRUE(
+        PlaceholderBindings::compare(&originalBindings, &optimizedBindings));
+  }
+
+  ExecutionEngine EE_;
+  Module *mod_{nullptr};
+  Function *originalF_{nullptr}, *optimizedF_{nullptr};
```

Comment:
I'd keep the original function as just `F_`. This is more in line with other tests and also the amount of changes in your PR.

Also, please add Doxygen comments for `originalF_`(or `F_`) and `optimizedF_`.